client/src/App.js

```diff
@@ -3,23 +3,47 @@ import { MuiThemeProvider } from "@material-ui/core";
 import { BrowserRouter, Route, Switch } from "react-router-dom";
 
 import { theme } from "themes/theme";
+
+import Navbar from "components/Navbar";
+import ProtectedRoute from "common/ProtectedRoute";
 import LoginPage from "pages/Login";
 import SignUp from "pages/SignUp";
+import Home from "pages/Home";
+import Profile from "pages/Profile";
+
+import { Provider as AuthProvider } from "contexts/AuthContext";
 
 import "App.css";
 
 function App() {
   return (
     <MuiThemeProvider theme={theme}>
       <BrowserRouter>
-        <Switch>
-          <Route path="/login" component={LoginPage} />
-          <Route path="/signup" component={SignUp} />
-          <Route path="/" component={SignUp} />
-        </Switch>
+        <AuthProvider>
+          <Switch>
+            <Route path="/login" render={(props) => <LoginPage {...props} />} />
+            <Route path="/signup" render={(props) => <SignUp {...props} />} />
+            <ProtectedRoute
+              path="/"
+              render={(props) => <LoggedInContainer {...props} />}
```

Comment:
all non-protected routes can **immediately** go back to simplified `<Route path="/login" component={LoginPage} />`
no need for the render function.  
the protected route can also follow same format of `<ProtectedRoute path="/" component={LoggedInContainer} />`
To make it work with protected route, you will need to make changes in my comment in `common/protectedroute.js`